src/components/File.vue

```diff
@@ -33,27 +33,41 @@
 			<div class="file__images">
 				<VideoIcon v-if="file.mime.includes('video')" class="video-icon" :size="64" />
 
-				<img v-if="visibility !== 'none' && canLoad && !errorNear && !loadedVisible"
-					ref="imgNear"
-					:key="`${file.basename}-near`"
-					:src="srcNear"
-					:alt="file.basename"
-					:aria-describedby="ariaDescription"
-					@load="onLoadNear"
-					@error="onErrorNear">
-
-				<img v-if="(visibility === 'visible' || (loadedVisible && visibility === 'near')) && canLoad && !errorVisible"
-					ref="imgVisible"
-					:key="`${file.basename}-visible`"
-					:src="srcVisible"
-					:alt="file.basename"
-					:aria-describedby="ariaDescription"
-					@load="onLoadVisible"
-					@error="onErrorVisible">
+				<!-- We have two img elements to load the small and large preview -->
+				<!-- Do not show the small preview if the larger one is loaded -->
+				<!-- Prioritize visible files -->
+				<!-- Load small preview first, then the larger one -->
+				<!-- Preload large preview for near visible files -->
+				<!-- Preload small preview for further away files -->
+				<template v-if="initialized">
+					<img v-if="!loadedLarge && (loadedSmall || (distance < 5 && !errorSmall))"
+						ref="imgSmall"
+						:key="`${file.basename}-small`"
+						:src="srcSmall"
+						:alt="file.basename"
+						:aria-describedby="ariaDescription"
+						:decoding="loadedSmall || isVisible ? 'sync' : 'async'"
+						:fetchpriority="loadedSmall || isVisible ? 'high' : 'low'"
+						:loading="loadedSmall || isVisible ? 'eager' : distance < 2 ? 'auto' : 'lazy'"
+						@load="onLoadSmall"
+						@error="onErrorSmall">
+
+					<img v-if="loadedLarge || ((isVisible || (distance < 2 && (loadedSmall || errorSmall))) && !errorLarge)"
+						ref="imgLarge"
+						:key="`${file.basename}-large`"
+						:src="srcLarge"
+						:alt="file.basename"
+						:decoding="loadedLarge || isVisible ? 'sync' : 'async'"
+						:fetchpriority="loadedLarge || isVisible ? 'high' : 'low'"
+						:loading="loadedLarge || isVisible ? 'auto' : 'lazy'"
+						:aria-describedby="ariaDescription"
+						@load="onLoadLarge"
+						@error="onErrorLarge">
```

Comment:
Looks like there might be room for a dedicated component or unifying those lines? :)

Reply:
I which, but they are slightly different and also dependent on each other, so an abstraction layer would bring unnecessary complexity. Also, in the current state, it is easily tweakable, and easier to read.